rocAL/include/api/rocal_api_data_loaders.h

```diff
@@ -824,4 +824,52 @@ extern "C" RocalTensor ROCAL_API_CALL rocalJpegExternalFileSource(RocalContext p
+                                                            bool shuffle = false,
+                                                            bool loop = false,
+                                                            bool downmix = false,
+                                                            unsigned max_frames = 1,
+                                                            unsigned max_channels = 1);
+
+/// Creates Audio file reader and decoder. It allocates the resources and objects required to read and decode audio files stored on the file systems. It has internal sharding capability to load/decode in parallel is user wants.
+/// If the files are not in standard audio compression formats they will be ignored.
+/// \param context Rocal context
+/// \param source_path A NULL terminated char string pointing to the location on the disk
+/// \param shard_id Shard id for this loader
+/// \param shard_count Defines the parallelism level by internally sharding the input dataset and load/decode using multiple decoder/loader instances. Using shard counts bigger than 1 improves the load/decode performance if compute resources (CPU cores) are available.
+/// \param is_output Determines if the user wants the loaded audio to be part of the output or not.
+/// \param shuffle Determines if the user wants to shuffle the dataset or not.
+/// \param loop Determines if the user wants to indefinitely loops through audio or not.
+/// \param downmix If set to True, downmix all input channels to mono. If downmixing is turned on, the decoder output is 1D. If downmixing is turned off, it produces 2D output with interleaved channels incase of multichannel audio.
+/// \param max_frames The maximum frames of the decoded audio.
+/// \param max_channels The maximum channels of the decoded audio.
+/// \param storage_type Determines the storage type
+/// \return Reference to the output audio
+extern "C"  RocalTensor  ROCAL_API_CALL rocalAudioFileSourceSingleShard(RocalContext p_context,
+                                                                        const char* source_path,
+                                                                        unsigned shard_id,
+                                                                        unsigned shard_count,
+                                                                        bool is_output,
+                                                                        bool shuffle = false,
+                                                                        bool loop = false,
+                                                                        bool downmix = false,
+                                                                        unsigned max_frames = 1,
+                                                                        unsigned max_channels = 1,
+                                                                        unsigned storage_type = 9);
```

Comment:
Why storage type is included here?

Reply:
So we need not ask user to pass storage type
You can see if text file is passed inside the API in rocal_api_dataloaders.cpp and you can set the storage type
Because to the end use the storage type is confusing